### PR TITLE
[qtmultimedia] Remove dependency alsa

### LINKED ISSUE #24875: [qtmultimedia] Build error
**Host Environment**
- OS: CentOS 7.6 (WSL2)/ Fedora 35 (WSL2)
- Compiler: GCC 11

**To Reproduce**
```bash
./vcpkg install qtmultimedia:x64-linux-dynamic  --overlay-ports=/opt/vcpkg-overlay/ports/147027390bd5bb13ce14e0bfc153873e424b71b2/ports #Qt 6.2.4
./vcpkg install qtmultimedia  --overlay-ports=/opt/vcpkg-overlay/ports/147027390bd5bb13ce14e0bfc153873e424b71b2/ports #Qt 6.2.4
```

**Failure logs**
```log
CMake Error at /opt/vcpkg/scripts/buildsystems/vcpkg.cmake:824 (_find_package):
  Could not find a package configuration file provided by "ALSA" with any of
  the following names:

    ALSAConfig.cmake
    alsa-config.cmake

  Add the installation prefix of "ALSA" to CMAKE_PREFIX_PATH or set
  "ALSA_DIR" to a directory containing one of the above files.  If "ALSA"
  provides a separate development package or SDK, be sure it has been
  installed.
Call Stack (most recent call first):
  /opt/vcpkg/installed/x64-linux-dynamic/share/Qt6/QtFindPackageHelpers.cmake:81 (find_package)
  src/multimedia/configure.cmake:9 (qt_find_package)
  /opt/vcpkg/installed/x64-linux-dynamic/share/Qt6/QtModuleHelpers.cmake:515 (include)
  src/multimedia/CMakeLists.txt:15 (qt_internal_add_module)
```

**Additional context**
I found that alsa is installed. I tried with CLion and 
```log
-- The C compiler identification is GNU 11.2.1
-- The CXX compiler identification is GNU 11.2.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /opt/rh/devtoolset-11/root/usr/bin/gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /opt/rh/devtoolset-11/root/usr/bin/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found ALSA: /usr/lib64/libasound.so (found version "1.1.8") 
-- Configuring done
-- Generating done
```

### PR DESCRIPTION
**Pull request by JackBoosY**
Remove alsa since it has some bugs.

Fixes #24875.

### CODE REVIEW

File: ports/qtmultimedia/portfile.cmake [hunk 1]
```diff
@@ -40,7 +40,6 @@ endif()
 
 if(VCPKG_TARGET_IS_LINUX)
     list(APPEND FEATURE_OPTIONS "-DFEATURE_alsa=ON")
-     list(APPEND FEATURE_OPTIONS "-DCMAKE_REQUIRE_FIND_PACKAGE_ALSA=ON")
```
Comment (Neumann-A) [inline ports/qtmultimedia/portfile.cmake:43]: why do you keep `-DFEATURE_alsa=ON` but deactivate `-DCMAKE_REQUIRE_FIND_PACKAGE_ALSA=ON`.
If you do that with `-DFEATURE_alsa=OFF` i would understand that.

Comment (Neumann-A) [reply ports/qtmultimedia/portfile.cmake:43]: Nevermind read the issue

Comment (JackBoosY) [reply ports/qtmultimedia/portfile.cmake:43]: So we have to remove dependency alsa, right?

Comment (Neumann-A) [reply ports/qtmultimedia/portfile.cmake:43]: na the problem is qt does `find_package(ALSA CONFIG)` before doing `find_package(ALSA)`

**Comment by JackBoosY** [comment]
@Neumann-A Now I met some compiler error:
```
home/work/vcpkg/buildtrees/qtmultimedia/src/here-src-6-14cbb5d7e0/src/multimedia/platform/qplatformmediaintegration.cpp: In static member function ‘static QPlatformMediaIntegration* QPlatformMediaIntegration::instance()’:
/home/work/vcpkg/buildtrees/qtmultimedia/src/here-src-6-14cbb5d7e0/src/multimedia/platform/qplatformmediaintegration.cpp:102:52: error: invalid new-expression of abstract class type ‘QAlsaIntegration’
  102 |             holder.nativeInstance.storeRelease(new PlatformIntegration);
      |                                                    ^~~~~~~~~~~~~~~~~~~
In file included from /home/work/vcpkg/buildtrees/qtmultimedia/x64-linux-dbg/include/QtMultimedia/6.3.1/QtMultimedia/private/qalsaintegration_p.h:1,
                 from /home/work/vcpkg/buildtrees/qtmultimedia/src/here-src-6-14cbb5d7e0/src/multimedia/platform/qplatformmediaintegration.cpp:54:
/home/work/vcpkg/buildtrees/qtmultimedia/x64-linux-dbg/include/QtMultimedia/6.3.1/QtMultimedia/private/../../../../../../src/here-src-6-14cbb5d7e0/src/multimedia/platform/alsa/qalsaintegration_p.h:60:7: note:   because the following virtual functions are pure within ‘QAlsaIntegration’:
   60 | class QAlsaIntegration : public QPlatformMediaIntegration
      |       ^~~~~~~~~~~~~~~~
In file included from /home/work/vcpkg/buildtrees/qtmultimedia/src/here-src-6-14cbb5d7e0/src/multimedia/platform/qplatformmediaintegration.cpp:41:
/home/work/vcpkg/buildtrees/qtmultimedia/src/here-src-6-14cbb5d7e0/src/multimedia/platform/qplatformmediaintegration_p.h:90:39: note:      ‘virtual QPlatformMediaFormatInfo* QPlatformMediaIntegration::formatInfo()’
   90 |     virtual QPlatformMediaFormatInfo *formatInfo() = 0;

```
Any ideas?

**Comment by JackBoosY** [comment]
@0x8A63F77D @m-kuhn Can you please test this PR whether it works?

**Comment by Neumann-A** [comment]
> @Neumann-A Now I met some compiler error:

deactivate ALSA then. Seems to be not ready yet. Also has moved paths in Qt 6.4

**Review by LilyWangLL — APPROVED** [review]

**Comment by m-kuhn** [comment]
I guess the testing is obsolete in this case?

**Comment by JackBoosY** [comment]
@m-kuhn Yeah... I've removed this dependency, so the issue will must be disappeared.

**Comment by m-kuhn** [comment]
That's similar enough to my current approach to assume it's working :+1:

**Comment by ras0219-msft** [comment]
SGTM for now, hope to get it reenabled in the future :). Thanks everyone!

**Pull request merged**